docker/daemon.go

```diff
@@ -150,16 +148,46 @@ func mainDaemon() {
 	job.Setenv("TlsCert", *flCert)
 	job.Setenv("TlsKey", *flKey)
 	job.SetenvBool("BufferRequests", true)
-	err := job.Run()
+
+	// The serve API job never exits unless an error occurs
+	// We need to start it as a goroutine and wait on it so
+	// daemon doesn't exit
+	serveAPIWait := make(chan error)
+	go func() {
+		if err := job.Run(); err != nil {
+			log.Errorf("ServeAPI error: %v", err)
```

Comment:
as per our discussion, see my last comment in the PR as to the "why", and this gist for example of output when there is duplicate logging (not **too** bad IMO): https://gist.github.com/estesp/c0f9f26852e9af4cbf4a